### ISSUE TITLE
Fix auth library integration tests to work in GHA

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#21]
One of the auth library tests is really an integration test, it depends on things in the environment:

https://github.com/googleapis/google-cloud-rust/blob/1e1ed307e39c16c18cfaf3895ec16d4138349353/auth/src/lib.rs#L354-L362

This test breaks under GHA.  I do not know exactly what the problem might be.  I am going to skip the test when GCE is not detected.  That will unblock other development, and we can fix auth later.

**Comment by coryan** [comment]
If you have Docker installed in your workstation you can repro locally using:

```bash
docker run --user $UID --workdir /workspace  --volume $PWD:/workspace  \
    --rm -it rust:latest cargo test --package google-cloud-auth
```

**Comment by dangazineu** [comment]
This is expecting you'll have application default credentials available. One way of fixing this, when running tests on the local environment, it to run `gcloud auth application-default login` as described [here](https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login).

I think we can add that step in the contributor guide as well.
